File: components/service-standard/src/com/github/joerodriguez/sbng2ex/web/ServiceResponseWebResponder.java
package com.github.joerodriguez.sbng2ex.web;

import com.github.joerodriguez.sbng2ex.service.ServiceResponse;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

public class ServiceResponseWebResponder {
    public static <T> ResponseEntity<T> respond(
            ServiceResponse<T> response,
            HttpStatus succcessStatusCode,
            HttpStatus failureStatusCode
    ) {
        if (!response.isSuccess()) {
            throw new ServiceFailureError(response.getErrors(), failureStatusCode);
        }

        return new ResponseEntity<>(response.getEntity(), succcessStatusCode);
    }
}

File: components/service-standard/src/com/github/joerodriguez/sbng2ex/web/ServiceResponseWebResponder.kt
package com.github.joerodriguez.sbng2ex.web

import com.github.joerodriguez.sbng2ex.service.ServiceResponse
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity

object ServiceResponseWebResponder {

    fun <T> respond(
            response: ServiceResponse<T>,
            succcessStatusCode: HttpStatus,
            failureStatusCode: HttpStatus
    ) : ResponseEntity<T> {
        if (!response.isSuccess) {
            throw ServiceFailureError(response.getErrors(), failureStatusCode)
        }

        return ResponseEntity<T>(response.entity, succcessStatusCode)
    }
}
